.github/workflows/unit.yaml

```diff
@@ -0,0 +1,30 @@
+name: Unit Tests
+on:
+  push:
+    paths-ignore:
+      - 'docs/**'
+  pull_request:
+  workflow_dispatch:
+
+jobs:
```

Comment:
Maybe we can use a matrix here to speed-up the workflow?

Reply:
I've figured out how to do this, about to push, but one thing I'm concerned about with this is that the UI will show all of the jobs and it can get kinda messy with 8+ jobs. Especially since the functional tests will have a job for each benchmark, this may get crazy! If it's something we are ok with though then that's great!